python/pynessie/model.py

```diff
@@ -276,9 +276,9 @@ def get_obj_type(self, obj: Operation) -> str:
 class CommitMeta:
     """Dataclass for commit metadata."""
 
-    hash_: str = desert.ib(fields.Str(data_key="hash"), default=None)
-    commitTime: datetime = desert.ib(fields.DateTime(), default=None)
-    authorTime: datetime = desert.ib(fields.DateTime(), default=None)
+    hash_: str = desert.ib(fields.Str(data_key="hash"), default="")
+    commitTime: datetime = desert.ib(fields.DateTime(), default=datetime.now())
+    authorTime: datetime = desert.ib(fields.DateTime(), default=datetime.now())
     committer: str = attr.ib(default=None, metadata=desert.metadata(fields.Str(allow_none=True)))
```

Comment:
```
  pynessie/model.py:279: error: Incompatible types in assignment (expression has type "None", variable has type "str")
  pynessie/model.py:280: error: Incompatible types in assignment (expression has type "None", variable has type "datetime")
  pynessie/model.py:281: error: Incompatible types in assignment (expression has type "None", variable has type "datetime")
```
for the problematic fields, have we tried using the same pattern as we do here in line 282?
i.e. `attr.ib(default=None, metadata=xyz)` ?
according to https://desert.readthedocs.io/en/stable/usage.html#use-with-attrs this would just be the "verbose form"

side note: also kind of surprising that in line 286 we are not hitting this issue because the `dict` type should also not be able to hold `None` as default.

Reply:
following my original suggestion and changing these 3 lines to:
```
    hash_: str = attr.ib(default=None, metadata=desert.metadata(fields.Str(data_key="hash")))
    commitTime: datetime = attr.ib(default=None, metadata=desert.metadata(fields.DateTime()))
    authorTime: datetime = attr.ib(default=None, metadata=desert.metadata(fields.DateTime()))
```
passed both linting and tests locally, can you confirm?
what errors are you getting exactly?

Reply:
This works for me as well. I still tried to use `desert.ib` which can be replaced safely by `attr.ib`.